terraform/variables.tf

```diff
@@ -54,20 +54,20 @@ variable "enable_autopilot" {
 
 variable "gke_node_pool" {
   type = object({
-    machine_type = string
-
-    oauth_scopes = list(string)
-
-    labels = map(string)
-
     initial_node_count = number
+    labels             = map(string)
+    machine_type       = string
 
     autoscaling = object({
-      min_node_count = number
       max_node_count = number
+      min_node_count = number
     })
   })
-  nullable    = true
   description = "Defines machine type and size of the default GKE cluster node pool"
-  default     = null
+  default = {
+    initial_node_count = 3
+    labels             = {}
+    machine_type       = "e2-medium"
```

Comment:
**Suggestion:**
I think the default node pool should be equivalent to the node pool used in [Online Boutique's quickstart](https://github.com/GoogleCloudPlatform/microservices-demo#quickstart-gke).

When I `terraform apply`-ed with 3 `e2-medium` Nodes, `shippingservice` said `Outofcpu`:

![Screenshot 2023-03-02 at 9 33 58 AM](https://user-images.githubusercontent.com/10292865/222458095-b6eb0793-53e4-4742-abdc-f01ce021f16a.png)
I think we'd at least need 4 Nodes if we're using `e2-medium`.
I'm guessing this was a one-off error since this Node count and machine type worked for you.

But I worry that a total of 6 CPUs (in the cluster) and 12 GB memory won't be enough when users try to add Istio/ASM/more deployments into their cluster. So let's use the same `machine_type` as the quickstart:
```suggestion
    machine_type       = "e2-standard-2"
```

Snippet from `gcloud compute machine-types list`:
```
NAME: e2-medium
CPUS: 2
MEMORY_GB: 4.00

NAME: e2-standard-2
CPUS: 2
MEMORY_GB: 8.00
```

Reply:
Obsolete due to remove of the customization feature.